File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int i=0;
        int j=nums.length;
        int sum=0;
        int max=Integer.MIN_VALUE;
        while(i<j){
            if(sum<0 && sum<nums[i]){
                sum=0;
            }
            sum+=nums[i];
            max=Math.max(sum,max);
            i++;
        }
        return max;
    }
}

File: 53-maximum-subarray/maximum-subarray.c
int maxSubArray(int* nums, int n) {
    int i = 0,j = n;
    int sum = 0;
    int max = INT_MIN;
    while(i<j){
        if(sum<0 && nums[i]>sum){
            sum=0;
        }
        sum+=nums[i];
        if(max<sum){
            max = sum;
        }
        i++;
    }
    return max;
}